samples/SimpleCustomConnectionPlugin.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * AWS JDBC Driver for MySQL
+ * Copyright Amazon.com Inc. or affiliates.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+import java.lang.reflect.InvocationTargetException;
+import java.sql.ResultSet;
+import java.util.concurrent.Callable;
+
+/**
+ * This is a simple example for writing a custom connection plugin.
+ * <p>
+ * All connection plugins must implement the {@link IConnectionPlugin} interface,
+ * so they can be automatically registered in the driver.
+ * Once registered, every connection will create an instance of this connection plugin.
+ */
+public class SimpleCustomConnectionPlugin implements IConnectionPlugin {
+
+  /**
+   * All method calls related to the connection object will be passed to this method as {@code Callable<?> executeSqlFunc}.
+   * This includes methods that may be called frequently, such as:
+   * <ul>
+   *   <li>{@link ResultSet#next()}</li>
+   *   <li>{@link ResultSet#getString(int)}</li>
+   * </ul>
+   * It is recommended to NOT add logs in this method to reduce performance overhead.
+   */
+  @Override
+  public Object execute(Class<?> methodInvokeOn, String methodName, Callable<?> executeSqlFunc) throws Exception {
+    try {
+      // Invoking `executeSqlFunc` may return an InvocationTargetException that needs to be unwrapped,
+      // so the underlying exception may be handled correctly in higher levels.
+      return executeSqlFunc.call();
+    } catch (InvocationTargetException invocationTargetException) {
```

Comment:
I think this exception has already been unwrapped in DefaultConnectionPlugin

Reply:
Let's add some "valuable" code to this example. For example, let's calculate method execution time and print in out to log.

Reply:
I thought about doing that but logging everything is heavy and we mentioned in our comments that we don't recommend adding logs in this metho